tpke/src/lib.rs

```diff
@@ -492,16 +476,158 @@ mod tests {
 
         // Malformed the ciphertext
         ciphertext.ciphertext[0] += 1;
-        let result = std::panic::catch_unwind(|| {
+        let result = panic::catch_unwind(|| {
             checked_decrypt_with_shared_secret(&ciphertext, aad, &shared_secret)
         });
         assert!(result.is_err());
 
         // Malformed the AAD
         let aad = "bad aad".as_bytes();
-        let result = std::panic::catch_unwind(|| {
+        let result = panic::catch_unwind(|| {
             checked_decrypt_with_shared_secret(&ciphertext, aad, &shared_secret)
         });
         assert!(result.is_err());
     }
+
+    #[test]
+    fn simple_threshold_decryption_with_share_refreshing_at_point() {
```

Comment:
I think the name "share refresh" here is confusing. We need to agree on a naming convention. Let's assume `N` is the original number of shares, and `t` is the threshold. There are 2 different types of share calculation protocols, used in 3 different ways:
1. `Ñ` parties (where `t <= Ñ <= N`) jointly execute a "share refresh" or "share renewal" algorithm, and the output is `M` new shares (with `M <= Ñ`), with each of the `M` new shares substituting the original share (i.e., the original share is deleted). I personally like "share refresh" but the original paper calls it "share renewal" (also fine).
2. `Ñ` parties (where `t <= Ñ <= N`) jointly execute a "share recovery" algorithm, and the output is 1 new share. There are 2 potential variants here:
    2.1. The new share is intended to restore a previously existing share, e.g., due to loss or corruption. It could also be used to on-board a new participant but reusing a previously existing share (sounds like a bad idea, though)
    2.2. The new share is independent from the previously existing shares. We can use this to on-board a new participant  into an existing cohort. 

In this test, you're actually doing 2.1, but the name "share refresh" is confusing to me. Let's settle in a convention for naming algorithms 1, 2.1 and 2.2 and use it consistently.